### PR TITLE
Fix SDL sends incorrect path for DYNAMIC image with AddSubMenu

### PR DESCRIPTION
**Pull request by indlu**
Fixed SDL sends incorrect path for DYNAMIC image with AddSubMenu

### CODE REVIEW

**Comment by indlu** [comment]
Please review @dcherniev, @okozlovlux , @LuxoftAKutsan

File: src/components/application_manager/include/application_manager/commands/mobile/add_sub_menu_request.h [hunk 1]
```diff
@@ -93,6 +93,17 @@ class AddSubMenuRequest : public CommandRequestImpl {
    */
   bool CheckMenuIconExistedInStorage();
 
+  /*
+   * @brief Add image file full path
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/mobile/add_sub_menu_request.h]: @indlu completely poor discription

File: src/components/application_manager/include/application_manager/commands/mobile/add_sub_menu_request.h [hunk 2]
```diff
@@ -93,6 +93,17 @@ class AddSubMenuRequest : public CommandRequestImpl {
...
+   *
+   * @param SmartObject with image
+   *
+   * @param app current application
+   */
+  static void ImageFullPathToMsgParams(smart_objects::SmartObject& image,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/mobile/add_sub_menu_request.h]: @indlu is htere any output parameters?

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc [hunk 3]
```diff
@@ -94,18 +94,49 @@ void AddSubMenuRequest::Run() {
...
+        full_file_path = file_system::CurrentWorkingDirectory() + "/" +
+                         app_storage_folder + "/";
+      }
+    } else {  // empty app storage folder
+      full_file_path = file_system::CurrentWorkingDirectory() + "/";
+    }
+    full_file_path += app->folder_name();
+    full_file_path += "/";
+    full_file_path += file_name;
+    image[strings::value] = full_file_path;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @indlu is it end of function or some errors in firmating?

File: src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc [hunk 4]
```diff
@@ -291,6 +291,10 @@ TEST_F(AddSubMenuRequestTest, Run_SUCCESS) {
       "valid_sub_menu_name";
   (*command_msg_)[am::strings::msg_params][am::strings::position] =
       "test_position";
+  (*command_msg_)[am::strings::msg_params][am::strings::sub_menu_icon]
+                 [am::strings::value] = "icon.png";
+  (*command_msg_)[am::strings::msg_params][am::strings::sub_menu_icon]
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc]: @indlu does this test actually check it's logic?

File: src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc [hunk 5]
```diff
@@ -94,18 +94,49 @@ void AddSubMenuRequest::Run() {
...
+        application_manager_);
     msg_params[strings::sub_menu_icon] =
         (*message_)[strings::msg_params][strings::sub_menu_icon];
+    SendHMIRequest(hmi_apis::FunctionID::UI_AddSubMenu, &msg_params, true);
+  } else {
+    LOG4CXX_ERROR(logger_, "Sub-menu icon is not valid.");
+    SendResponse(false, mobile_apis::Result::INVALID_DATA);
+  }
+}
+void AddSubMenuRequest::ImageFullPathToMsgParams(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc]: @indlu please avoid passing smart object. Use just string as return value.

File: src/components/application_manager/include/application_manager/commands/mobile/add_sub_menu_request.h (hunk 1 above)
Comment (indlu) [inline src/components/application_manager/include/application_manager/commands/mobile/add_sub_menu_request.h]: @LuxoftAKutsan, fixed in 4654e74

**Comment by indlu** [comment]
@LuxoftAKutsan, fixed in 4654e74

**Pull request merged**